crates/bevy_reflect/Cargo.toml

```diff
@@ -34,6 +34,9 @@ bevy_reflect_derive = { path = "derive", version = "0.15.0-dev" }
 bevy_utils = { path = "../bevy_utils", version = "0.15.0-dev" }
 bevy_ptr = { path = "../bevy_ptr", version = "0.15.0-dev" }
 
+# used by bevy-utils, but it also needs reflect impls
+foldhash = { version = "0.1.3", default-features = false }
+
```

Comment:
This felt like the best approach: rather than re-exporting foldhash in `bevy_utils` and further bloating up that crate, just re-import it here and add impls for it.

Reply:
My only critique here would be to make the `foldhash` dependency optional, keeping inline with how other reflection implementations for 3rd party types are done. Not a blocker, just a comment.

Reply:
I didn't make it optional only because it isn't actually optional: `bevy_utils` always includes it, so, putting it optional here would just give people the option to accidentally remove impls for it when they need them.